pkg/reconciler/route/resources/ingress.go

```diff
@@ -193,9 +193,18 @@ func makeIngressSpec(
 		}
 	}
 
+	var httpOption netv1alpha1.HTTPOption
+	if config.FromContext(ctx).Network.HTTPProtocol == network.HTTPEnabled {
+		httpOption = netv1alpha1.HTTPOptionEnabled
+	}
+	if config.FromContext(ctx).Network.HTTPProtocol == network.HTTPRedirected {
+		httpOption = netv1alpha1.HTTPOptionRedirected
+	}
```

Comment:
If someone uses `Disabled` currently, it will be set to `Enabled` silently. Is that alright? cc @ZhiminXiang

Reply:
Good question. I think we may want to mark the Ingress as Failed when Disabled is setting so that users will be aware of that we no longer support Disabled.

Reply:
Hmm... If we stop supporting Disabled, it should have "deprecated period".
Making the Ingress as Failed when Disabled, it may break the services during upgrade :thinking:

Reply:
Currently there is no default setting for `httpOption` when it is empty.

I think In the short term, if the configmap is set to Disabled, we can just leave `httpOption` in the Ingress as empty. If the ingress implementation already consumed the HTTPProtocol in the ConfigMap, they should keep consuming it when the `httpOption` is empty for the compatibility consideration.

In the meantime, I think we should start deprecating Disabled in the config-network ConfigMap. The Ingress implementations can still choose to keep this option by adding Disabled to the implementation specific ConfigMap if they want.

Reply:
Sounds good to me. Thank you!

Reply:
Have made some changes to set it to empty for Disabled